.github/workflows/windows.yml

```diff
@@ -0,0 +1,50 @@
+name: Build Windows release
+
+on:
+  push:
+    branches: [ "main", "ci-*"]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-windows-release:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.12']
+      fail-fast: true
+
+    steps:
+    - uses: actions/checkout@v4.1.2
+      with:
+        fetch-depth: 0
+
+    - name: Get short commit ref
+      run: |
+        version=$(
+            git describe --tags --dirty \
+            | sed -e 's/^v//' -e 's/\([^-]*-g\)/r\1/;s/-/./g'
+          )
+        echo "PAPIS_VERSION=$version" >> "$GITHUB_ENV"
+        
+
+    - name: Set version in pyproject.toml
+      run: |
+        set -x \
+        sed -i \
+          "s/^version =/version = \"${{env.PAPIS_VERSION}}\"/" \
+          pyproject.toml
+    
+    - name: Build Windows binary
+      run: |
+        docker run -v ${{github.workspace}}:/src \
+         -e PIP_GROUPS=windows,develop \
+         -e SPECFILE=scripts/windows/papis.spec \
+         ghcr.io/kiike/pyinstaller-windows:latest
+    
+    - name: Publish binary as artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: papis-${{env.PAPIS_VERSION}}
+        path: ${{ github.workspace }}/dist/papis.exe
```

Comment:
Then if I understand you correctly, what I should change is the events. So build only on tags creation or releases, right?

Reply:
Yeah, that's what I had in mind, but I was mostly thinking out loud.

Where are those artifacts going to go? They're not very discoverable on their own.

Reply:
The artifacts stay there. Then we could link them to the release assets.

Reply:
Hm, CI logs go away after a maximum of 90 days, I think. Do they artifacts remain?

Anyways, I'm fine with running this on every pull request too, to make sure it keeps working. Maybe just add an `if` on the publish step
```
    - if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```
so that artifacts are only created for tags, but the rest of the build process runs every time?

Reply:
By default, yes, they would stay there 90 days (by default)  in the artifacts storage. But of course we could move them somewhere else too.

I could apply that line but I fail to see why you would build without publishing the artifact though; if you want to see if the workflow works, I could apply a "ci" or "*windows*" tag event as well as pull request, like what was  [in the first commit](https://github.com/papis/papis/pull/795/commits/9edc953006a2d2d8764874bf7ff20a629e309eb4#diff-5ce5c9ff86f58b1f87b35b5227b2e84cb69f022d6741e1854f3e1e181091773cR3-R7) but editing the branch name.

Reply:
That works too! Thanks!